ui/src/Annotation.tsx

```diff
@@ -56,6 +51,20 @@ export class Annotation extends React.PureComponent<
     this.context.setSelectedAnnotationId(this.props.id);
   }
 
+  shouldHighlight() {
+    if (!this.context.selectedAnnotationId) { return false; }
+    const [typeSelected, idSelected,] = this.context.selectedAnnotationId.split('-');
+    const [typeSelf, idSelf,] = this.props.id.split('-');
+    if (typeSelf !== typeSelected) { return false; }
+    
+    // We are only focusing on symbols for right now;
+    if (typeSelf === 'symbol') {
+      const highlightedIds = this.context.symbolMatchSet[Number(idSelected)];
+      return this.isSelected() || highlightedIds.has(Number(idSelf));
+    }
+    return false;
+  }
```

Comment:
Can this code go in `SymbolAnnotation`? It feels like it might belong as logic for the symbol annotation component, rather than code that all annotations execute.

Reply:
Yes agreed that also makes more sense from an abstraction perspective as all annotations don't need to know how to highlight specific annotation types